x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/accesscontrol/IndicesAccessControl.java

```diff
@@ -154,23 +145,16 @@ public boolean hasDocumentLevelSecurity() {
      */
     public static class IndexAccessControl implements CacheKey {
 
-        private final boolean granted;
+        public static final IndexAccessControl ALLOW_ALL = new IndexAccessControl(null, null);
+
         private final FieldPermissions fieldPermissions;
         private final DocumentPermissions documentPermissions;
 
-        public IndexAccessControl(boolean granted, FieldPermissions fieldPermissions, DocumentPermissions documentPermissions) {
-            this.granted = granted;
```

Comment:
The heart of the change.

Reply:
In hindsight, when `granted` is `false`, both FieldPermissions and DocumentPermissions should be set to values that do *not* allow any fields or any documents. Having `granted=false` along with allowing all fields and documents was a terrible mental model.